File: example_bots/exampleBot.py
import random
import sys

def eprint(*args, **kwargs):
    print(*args, file=sys.stderr, **kwargs)

while(True):
    width, height = int(input()), int(input())
    map = [ [0]*width for i in range(height)]
    x = str()
    for h in range(int(height)):
        l = input()
        for w in range(width):
            map[h][w] = l[w]

    worm_position = [int(i) for i in input().split(' ')]
    priority, number_players = int(input()), int(input())
    my_position = [int (i) for i in input().split(' ')]
    other_players_positions = list()
    for j in range(int(number_players) - 1):
        s = input().split(' ')
        other_players_positions.append((s[0], [int (i) for i in s[1:]]))
    
    def test_move(move):
        if move == 'n':
            new_position = [my_position[0], my_position[1]-1]
        elif move == 's':
            new_position = [my_position[0], my_position[1]+1]
        elif move == 'e':
            new_position = [my_position[0]+1, my_position[1]]
        elif move == 'w':
            new_position = [my_position[0]-1, my_position[1]]
        else:
            new_position = my_position
        
        # check borders
        if new_position[0]<0 or new_position[1]<0 or new_position[0]>width or new_position[1]>height:
            return False
        
        map_symbol = map[new_position[1]][new_position[0]]
        # check holes
        if map_symbol == 'O':
            return False
        # check worm position
        if new_position==worm_position:
            return False
        
        return True
    
    moves = ['n', 's', 'e', 'w', '.']
    random.shuffle(moves)
    for m in moves:
        if test_move(m):
            print(m)
            break

File: example_bots/exampleBot.c
#include <stdio.h>
#include <stdlib.h>

#define TRUE 1
#define FALSE 0
#define eprintf(...) fprintf (stderr, __VA_ARGS__)

struct position {
    int x;
    int y;
    char id;
} typedef Position;

char moves[] = {'n', 's', 'e', 'w'};

char **map; // map coordinates are y, x
int width, height;

int test_move(char move, Position *my_position, Position *worm){
    int move_x = my_position->x, move_y = my_position->y;
    switch(move) {
        case 'n':
            move_y-=1;
            break;
        case 's':
            move_y+=1;
            break;
        case 'e':
            move_x+=1;
            break;
        case 'w':
            move_x-=1;
            break;
    }

    // check borders
    if (move_x<0 || move_y<0 || move_x>width || move_y>height){
        return FALSE;
    }

    char map_symbol = map[move_y][move_x];
    // check holes
    if (map_symbol == 'O'){
        return FALSE;
    }
    // check worm position
    if (move_x == worm->x && move_y == worm->y){
        return FALSE;
    }

    // good move
    return TRUE;
}

// shuffle moves array
void shuffleMoves() {
    for (int i = 0; i < 3; i++) 
    {
        int j = i + rand() / (RAND_MAX / (4 - i) + 1);
        char t = moves[j];
        moves[j] = moves[i];
        moves[i] = t;
    }
}

int main() {
    int priority;
    int num_players;
    Position worm_position;
    Position my_position;
    Position *players_positions;

    scanf("%d\n", &width);
    scanf("%d\n", &height);

    map = (char **) malloc(sizeof(char *) * height);
    for (int i = 0; i < height; i++) {
        map[i] = (char *)malloc(sizeof(char) * (width + 1));
        scanf("%s\n", map[i]);
    }
    scanf("%d %d\n", &worm_position.x, &worm_position.y);
    
    scanf("%d\n", &priority);
    scanf("%d\n", &num_players);
    players_positions = (Position *) malloc(sizeof(Position) * num_players);

    while (TRUE)
    {
        scanf("%d %d", &my_position.x, &my_position.y);
        for (int i = 0; i < num_players - 1; i++) {
            getchar();
            scanf("%c %d %d", &players_positions[i].id ,&players_positions[i].x, &players_positions[i].y);
        }

        shuffleMoves();
        char move = '.';
        for (int i=0; i<4; i++) {
            move = moves[i];
            if (test_move(move, &my_position, &worm_position)) {
                printf("%c\n", move);
                break;
            }
        }
        fflush(stdout);

        int ans = scanf("%d", &width);
        if (ans == EOF){
            break;
        }
        scanf("%d", &height);
        
        for (int i = 0; i < height; i++) {
            scanf("%s", map[i]);
        }
        scanf("%d %d", &worm_position.x, &worm_position.y);
        scanf("%d", &priority);
        scanf("%d", &num_players);

    }

    return 0;
}
